File: examples/javascript/read.js
const fs = require('node:fs');
const path = require('node:path');

const dataDirectory = path.join(__dirname, '..', '..', 'iraqi-cities-districts-data');
const cities = JSON.parse(
  fs.readFileSync(path.join(dataDirectory, 'cities.json'), 'utf8'),
);
const districts = JSON.parse(
  fs.readFileSync(path.join(dataDirectory, '1-NJF.json'), 'utf8'),
);

const najaf = cities.find((city) => city.key === 'NJF');

console.log(`${najaf.name} (${najaf.key})`);
console.log(`Districts: ${districts.length}`);
console.log('First district:', districts[0]);

File: examples/python/read.py
import json
from pathlib import Path


DATA_DIRECTORY = Path(__file__).parents[2] / "cites_distracts"


def load_json(filename):
    return json.loads((DATA_DIRECTORY / filename).read_text(encoding="utf-8"))


cities = load_json("cities.json")
districts = load_json("1-NJF.json")
najaf = next(city for city in cities if city["key"] == "NJF")

print(f'{najaf["name"]} ({najaf["key"]})')
print(f"Districts: {len(districts)}")
print("First district:", districts[0])